Zend/tests/enum/implements.phpt

```diff
@@ -0,0 +1,34 @@
+--TEST--
+Enum implements implemented in enum
+--FILE--
+<?php
+
+interface Colorful {
+    public function color(): string;
+}
+
+enum Suit implements Colorful {
```

Comment:
The existence of https://www.php.net/arrayaccess may introduce some edge cases to handle for constant expression evaluation.

It'd be inconvenient to start allowing constants that aren't actually constant

The most straightforward solution would be to forbid implementing ArrayAccess entirely, similar to how 
many magic methods are forbidden by this RFC (I'd recommend that). It may also be possible to allow it but check for mutable objects, references, or cyclical data structures

If something is already forbidding doing this, it'd be useful to add a test that the error is reported properly or the code behaves as expected.

```php
<?php
enum Foo implements ArrayAccess {
    case Bar;

    public function offsetExists($key) {}
    public function offsetGet($key) {
        return new stdClass();
    }
    public function offsetSet($key, $value) {}
    public function offsetUnset($key) {}
}
class X {
    const ACTUALLY_MUTABLE_OBJECT = Foo::BAR[0];
}
```

Reply:
Ugh, yeah this is a backdoor to evaluate anything in constant expressions. It does work apparently but probably not something we want to allow. We could probably throw error out if the object in `zend_ast_evaluate` -> `ZEND_AST_DIM` is an enum.

Reply:
That sounds like the best option to me - I don't know what type of objects zend_ast_evaluate is currently used with, but only forbidding enums won't break anything.

A third option is to "allow it but check for mutable objects, references, or cyclical data structures" but evaluating anything in constant expressions is extremely unpopular because of the potential for making code harder to understand, etc.

Reply:
Done